storage/mongodb/mongodb-test.yaml

```diff
@@ -0,0 +1,110 @@
+---
+- hosts: all
+  vars_files:
+    - external_vars.yaml
+  tasks:
+  - name: print hostname
+    debug:
+      msg: "hostname: {{ inventory_hostname }} and test_project_name: {{ test_project_name }}"
+    tags:
+      - setup
+
+  - name: clean tmp folder
+    file:
+      state: absent
+      path: "{{ tmp_folder }}/"
+    tags:
+      - setup
+
+  - name: copy files to remote
+    copy:
+      src: files
+      dest: "{{ tmp_folder }}"
+    tags:
+      - setup
+
+  - block:
+    - name: list all scripts
+      find:
+        paths: "{{ tmp_folder }}/files/scripts"
+        patterns: "*.sh"
+      register: tmp_script_glob
+
+    - name: make scripts executable
+      file:
+        path: "{{ item.path }}"
+        mode: 0744
+      with_items:
+        - "{{ tmp_script_glob.files }}"
+    tags:
+      - setup
+
+  - block:
+      - name: delete project
+        shell: oc delete project {{ test_project_name }}
+        ignore_errors: True
+      - name: wait until the project is gone
+        shell: oc get project {{ test_project_name }}
+        register: get_project_result
+        until: get_project_result.stderr.find(" not found") != -1
+        retries: 5
+        delay: 10
+        ignore_errors: True
+      - name: fail the test if the project is still there after 60s
+        fail:
+          msg: "the pod is NOT deleted after 60s"
+        when: get_project_result.rc == 0
+    when: delete_test_project_before_test|bool == true
+    tags:
+      - setup
+
+  - name: create test project
+    command:  oc new-project {{ test_project_name }}
+    tags:
+      - setup
+
+  - name: create mongodb
+    shell: oc process -f {{ tmp_folder }}/files/oc/mongodb-persistent-template.yaml -p MEMORY_LIMIT={{ MEMORY_LIMIT }} -p MONGODB_USER={{ MONGODB_USER }} -p MONGODB_PASSWORD={{ MONGODB_PASSWORD }} -p MONGODB_DATABASE={{ MONGODB_DATABASE }} -p VOLUME_CAPACITY={{ VOLUME_CAPACITY }} -p MONGODB_VERSION={{ MONGODB_VERSION }} -p STORAGE_CLASS_NAME={{ STORAGE_CLASS_NAME }} | oc create --namespace={{ test_project_name }} -f -
+    tags:
+      - setup
+
+  - block:
+      - name: wait until the mongodb pod is running
+        shell: oc get pod -n "{{ test_project_name }}" | grep -v deploy | grep mongodb | grep Running
+        register: pod_ready_result
+        until: pod_ready_result.rc == 0
+        retries: 5
+        delay: 10
+        ignore_errors: True
+      - name: fail the test if the pod is NOT running after 60s
+        fail:
+          msg: "the pod is NOT running after 60s"
+        when: pod_ready_result|failed
+    tags:
+      - setup
+
+  - name: create ycsb
+    shell: oc create -f {{ tmp_folder }}/files/oc/dc_ycsb.yaml
+    tags:
+      - setup
+
+  - block:
+      - name: wait until the ycsb pod is running
+        shell: oc get pod -n "{{ test_project_name }}" | grep -v deploy | grep ycsb | grep Running
+        register: pod_ready_result
+        until: pod_ready_result.rc == 0
+        retries: 5
```

Comment:
I would increase this number of retries to 10 - it failed for test case when image is not present

Reply:
Done